worlds/sc2/options.py

```diff
@@ -1644,6 +1648,30 @@ def get_excluded_missions(world: 'SC2World') -> Set[SC2Mission]:
     return excluded_missions
 
 
+def is_mission_in_soa_presence(
+    spear_of_adun_presence: int,
+    mission: SC2Mission,
+    option_class: Type[SpearOfAdunPresence] | Type[SpearOfAdunPassiveAbilityPresence] = SpearOfAdunPresence
+) -> bool:
+    """
+    Returns True if the mission can have Spear of Adun abilities.
+    No-build presence must be checked separately.
+    """
+    return (
+        (spear_of_adun_presence == option_class.option_everywhere)
+        or (spear_of_adun_presence == option_class.option_protoss and MissionFlag.Protoss in mission.flags)
+        or (spear_of_adun_presence == option_class.option_any_race_lotv
+            and (mission.campaign == SC2Campaign.LOTV or mission == SC2Mission.INTO_THE_VOID)
```

Comment:
I'd prefer `mission.campaign == SC2Campaign.LOTV or MissionFlag.VanillaSoa in mission.flags` instead (using flags instead of mentioning a specific mission)